common/src/main/java/com/wynntils/functions/generic/StringFunctions.java

```diff
@@ -282,4 +283,20 @@ public FunctionArguments.RequiredArgumentBuilder getRequiredArgumentsBuilder() {
                     new FunctionArguments.Argument<>("replacement", String.class, null)));
         }
     }
+
+    public static class IteratorFunction extends GenericFunction<String> {
```

Comment:
This is a very generic name for a very specific function. Maybe you should bite the bullet about it's specialized use, call it "ColorIterator", and let it return a full color code including the `§`?

Otherwise I think it should be at least renamed something more specific, like CharIterator or CharacterIterator or StringIterator.

Actually, even "iterator" is not ideal. It sounds like a programming construct.

Maybe `ColorProgression`?